### PR TITLE
Improve HexToPubkey & HexToSignature

### PR DESCRIPTION
**Pull request by jtraglia**
## 📝 Summary

This PR adds checks to `HexToPubkey` and `HexToSignature` to ensure they are valid points.

---

## ✅ I have run these commands

* [x] `make lint`
* [x] `make test`
* [x] `go mod tidy`

### CODE REVIEW

File: utils/utils_test.go [hunk 1]
```diff
@@ -35,18 +35,28 @@ func TestHexToPubkey(t *testing.T) {
 	_, err := HexToPubkey("0x01")
 	require.Error(t, err)
 
-	a, err := HexToPubkey("0xed7f862045422bd51ba732730ce993c94d2545e5db1112102026343904fcdf6f5cf37926a3688444703772ed80fa223f")
+	// This is an invalid pubkey
+	_, err = HexToPubkey("0xed7f862045422bd51ba732730ce993c94d2545e5db1112102026343904fcdf6f5cf37926a3688444703772ed80fa223f")
```
Comment (avalonche) [inline utils/utils_test.go:39]: non blocking: can be written as a table driven test to separate the test cases

File: utils/utils_test.go [hunk 2]
```diff
@@ -35,18 +35,28 @@ func TestHexToPubkey(t *testing.T) {
...
+	require.Equal(t, "0xac6e77dfe25ecd6110b8e780608cce0dab71fdd5ebea22a16c0205200f2f8e2e3ad3b71d3499c54ad14d6c21b41a37ae", a.String())
 }
 
 func TestHexToSignature(t *testing.T) {
 	_, err := HexToSignature("0x01")
 	require.Error(t, err)
 
-	a, err := HexToSignature("0xb8f03e639b91fa8e9892f66c798f07f6e7b3453234f643b2c06a35c5149cf6d85e4e1572c33549fe749292445fbff9e0739c78159324c35dc1a90e5745ca70c8caf1b63fb6678d81bd2d5cb6baeb1462df7a93877d0e22a31dd6438334536d9a")
+	// This is an invalid signature
+	_, err = HexToSignature("0xb8f03e639b91fa8e9892f66c798f07f6e7b3453234f643b2c06a35c5149cf6d85e4e1572c33549fe749292445fbff9e0739c78159324c35dc1a90e5745ca70c8caf1b63fb6678d81bd2d5cb6baeb1462df7a93877d0e22a31dd6438334536d9a")
```
Comment (avalonche) [inline utils/utils_test.go:53]: non blocking: same here as a table driven test

File: utils/utils_test.go (hunk 1 above)
Comment (jtraglia) [reply utils/utils_test.go:39]: Okay yeah good suggestion. I'll update this. And fix the linter issue :(

**Comment by jtraglia** [comment]
One minute. Adding some more tests.

**Review by metachris — APPROVED** [review]
nice, thanks!

**Pull request merged**